osbs/utils.py

```diff
@@ -327,20 +328,34 @@ def make_name_from_git(repo, branch, limit=53, separator='-', hash_size=5):
 
     size = (2 * len(separator)) + hash_size
 
+    disallowed = '._'
     for i in range(max(len(repo), len(branch))):
         for group, group_chars in groups:
             if i < len(group):
-                group_chars.append(group[i])
+                char = group[i]
+                if char in disallowed:
```

Comment:
I think it might be easier to understand the code if `repo` and `branch` are sanitized right after hash is computed:

```python
valid_chars = re.compile('[0-9a-z]')
repo = filter(valid_chars.match, repo)
branch = filter(valid_chars.match, branch)
```

Here's a sample:

```python
>>> import re
>>> repo = 'hello.world-bah!'
>>> valid_chars = re.compile('[0-9a-z]')
>>> filter(valid_chars.match, repo)
'helloworldbah'
```